### ISSUE TITLE
How to actually run on Ubuntu Linux? Including sdl?

### ISSUE THREAD

**Issue by jason-mehmel** [angband/angband#4745]
Hey there,

I'm on Ubuntu 20.04.

I've followed the compiling instructions here: https://angband.readthedocs.io/en/latest/hacking/compiling.html#id4

I've done:
```
./configure --with-no-install [other options as needed]
make
```

But I can't find an Angband file to run when it's done. Is it creating an executable that I can run from anywhere?

I do have Angband 3.5 installed from the Ubuntu repositories, so is the repository version getting priority?

I'm also not sure how to enable SDL with those commands.

**Comment by backwardsEric** [comment]
From the same directory where you ran make, you could run

`src/angband -mx11`

Or you could add

`make install`

when building.  There'll be a message about the installation failing because activation.txt is already there, but you can ignore that and run

`./angband -mx11`

If you built the SDL or SDL2 front ends and wanted to use one of them instead of the X11 one, replace '-mx11' in the above commands with '-msdl' or '-msdl2'.

**Comment by jason-mehmel** [comment]
> From the same directory where you ran make, you could run
> 
> `src/angband -mx11`
> 
> Or you could add
> 
> `make install`
> 
> when building. There'll be a message about the installation failing because activation.txt is already there, but you can ignore that and run
> 
> `./angband -mx11`
> 
> If you built the SDL or SDL2 front ends and wanted to use one of them instead of the X11 one, replace '-mx11' in the above commands with '-msdl' or '-msdl2'.

Do I also need to enable SDL in the ./configure command?

**Comment by backwardsEric** [comment]
Yes, the SDL front-end is disabled by default even if the necessary libraries are available.  Run

`./configure --with-no-install --enable-sdl`

to enable it before running make.

**Comment by jason-mehmel** [comment]
Okay, I'm trying to install with SDL but I'm getting a missing libraries error. Checking my system, I do have the following:

# The video subsystem requires libsdl2, libsdl2-ttf and libsdl2-image.
# The sound subsystem requires libsdl2 and libsdl2-mixer.

Is there anything here I'm missing?

**Comment by backwardsEric** [comment]
There's a different configure option for working with SDL2 (and the front-end is different than the one used for SDL1).  Configure for SDL2 with

`./configure --with-no-install --enable-sdl2`

After building, you can bring up Angband with the SDL2 front end by running

`src/angband -msdl2`

**Comment by jason-mehmel** [comment]
When I run that configure command, here's the SDL stuff that I see:

checking for sdl2-config... /usr/bin/sdl2-config
checking for SDL2 - version >= 2.0.0... yes
checking for IMG_LoadPNG_RW in -lSDL2_image... no
checking for TTF_Init in -lSDL2_ttf... yes
checking for Mix_OpenAudio in -lSDL2_mixer... yes

and:

-- Frontends --
- Curses                                  Yes
- X11                                     Yes
- SDL2                                    No; missing libraries
- SDL                                     Disabled
- Windows                                 Disabled
- Test                                    No
- Stats                                   No

- SDL2 sound                              Yes
- SDL sound                               Disabled


I'm not sure what libraries I could be missing!

**Comment by backwardsEric** [comment]
The key message there is 'checking for IMG_LoadPNG_RW in -lSDL2_image... no'.  In Debian, that's supplied by the lib-sdl2-image-dev package (and it's dependencies).  It looks like the required package in Ubuntu 20.04 goes by the same name.

**Comment by jason-mehmel** [comment]
Okay! We're getting closer!

Here's what I get when I try: src/angband -msdl2

src/angband: Unable to prepare any 'display module'!

**Comment by backwardsEric** [comment]
Did you rerun make after running ./configure (I'm suspecting src/angband is the one from the earlier build without SDL2 support)?  You can see what -m options the angband executable is willing to accept by running

`src/angband -?`

(any invalid option instead of -? would also work).

**Comment by jason-mehmel** [comment]
I tried re-running the configure command with the SDL2 and then make.

Here's what I'm getting now:

checking for sdl2-config... /usr/bin/sdl2-config
checking for SDL2 - version >= 2.0.0... yes
checking for IMG_LoadPNG_RW in -lSDL2_image... yes
checking for TTF_Init in -lSDL2_ttf... yes
checking for Mix_OpenAudio in -lSDL2_mixer... yes

And for frontends:

 -- Frontends --
- Curses                                  Yes
- X11                                     Yes
- SDL2                                    Yes
- SDL                                     Disabled
- Windows                                 Disabled
- Test                                    No
- Stats                                   No

- SDL2 sound                              Yes
- SDL sound                               Disabled

When I ran: `src/angband -?`

Usage: angband [options] [-- subopts]
  -n             Start a new character (WARNING: overwrites default savefile without -u)
  -l             Lists all savefiles you can play
  -w             Resurrect dead character (marks savefile)
  -g             Request graphics mode
  -x<opt>        Debug options; see -xhelp
  -u<who>        Use your <who> savefile
  -d<dir>=<path> Override a specific directory with <path>. <path> can be:
    scores (default is /home/jason/Downloads/Angband-4.2.2/lib/scores)
    gamedata (default is /home/jason/Downloads/Angband-4.2.2/lib/gamedata)
    screens (default is /home/jason/Downloads/Angband-4.2.2/lib/screens)
    help (default is /home/jason/Downloads/Angband-4.2.2/lib/help)
    info (default is /home/jason//.angband/Angband/info)
    pref (default is /home/jason/Downloads/Angband-4.2.2/lib/customize)
    fonts (default is /home/jason/Downloads/Angband-4.2.2/lib/fonts)
    tiles (default is /home/jason/Downloads/Angband-4.2.2/lib/tiles)
    sounds (default is /home/jason/Downloads/Angband-4.2.2/lib/sounds)
    icons (default is /home/jason/Downloads/Angband-4.2.2/lib/icons)
    user (default is /home/jason//.angband/Angband)
    save (default is /home/jason/Downloads/Angband-4.2.2/lib/save)
    archive (default is /home/jason/Downloads/Angband-4.2.2/lib/archive)
                 Multiple -d options are allowed.
  -s<mod>        Use sound module <sys>:
     sdl   SDL_mixer sound module
        
  -m<sys>        Use module <sys>, where <sys> can be:
     x11   Basic X11, subopts -d<display> -n<windows> -x<file>
     gcu   Text mode, subopts
              -B     Use brighter bold characters
              -D     Use terminal default background color
              -nN    Use N terminals (up to 6)

**Comment by jason-mehmel** [comment]
So it doesn't seem to show the SDL2 module option.

**Comment by backwardsEric** [comment]
Did you rerun make after running `./configure  --with-no-install --enable-sdl2`?  It shouldn't be necessary but running `make clean` and then `make install` would force it to rebuild everything.

**Comment by backwardsEric** [comment]
The "shouldn't be necessary" part in my last comment was wrong.  With the way the configure script and make files are currently, if you build and then rerun configure with different front ends enabled, you'll either need to run `make clean` or, to be minimalistic about what needs to be rebuilt, `rm src/main.o` before rebuilding.  Then you should see the effect of the different options to configure.

**Comment by jason-mehmel** [comment]
Okay! The `make clean` and then `make ` did the job, I can now run Angband with SDL2!

Which leads me to my next problem; I assumed the SDL2 version would include all the tiles, not to mention being able to add the extra windows, but none of that seems to work. Is there something extra I should have done to get that working?

Also, the SDL2 version opens in fullscreen, but even trying to use the menu doesn't let me leave the fullscreen view.

(If I should start a new issue for these requests, let me know!)

**Comment by backwardsEric** [comment]
I'd put those in a new issue and leave this one as a record of your experience with configure and getting the necessary dependencies for SDL2.

It's not in https://angband.readthedocs.io/en/latest , but a bit of documentation was added recently for how to set things with the different front ends.  You might look at that, https://github.com/angband/angband/blob/master/docs/customize.rst (skip to the "Interface details" and then "SDL2"; it's very close to the end of the file) and see if that helps with your questions.  For toggling fullscreen mode, selecting "Menu" then clicking on "Fullscreen" works for me (on Debian Buster).  For configuring tiles, selecting "Menu", "Angband", "Tiles", and then clicking on the tile set you want works for me (since the splash screen has no tiles; you'll likely want a character loaded to see the effect when switching tile sets).

**Comment by NickMcConnell** [comment]
Fixed in 1b862dc35
